presto-main/src/main/java/io/prestosql/util/Failures.java

```diff
@@ -94,9 +96,12 @@ private static ExecutionFailureInfo toFailure(Throwable throwable, Set<Throwable
         if (throwable instanceof PrestoTransportException) {
             remoteHost = ((PrestoTransportException) throwable).getRemoteHost();
         }
+        if (throwable instanceof SemanticException) {
+            semanticErrorCode = ((SemanticException) throwable).getCode();
+        }
 
         if (seenFailures.contains(throwable)) {
-            return new ExecutionFailureInfo(type, "[cyclic] " + throwable.getMessage(), null, ImmutableList.of(), ImmutableList.of(), null, GENERIC_INTERNAL_ERROR.toErrorCode(), remoteHost);
+            return new ExecutionFailureInfo(type, "[cyclic] " + throwable.getMessage(), null, ImmutableList.of(), ImmutableList.of(), null, GENERIC_INTERNAL_ERROR.toErrorCode(), remoteHost, semanticErrorCode);
```

Comment:
Nit: this argument list is getting long. Please wrap each argument like the call below.